synapse/handlers/saml_handler.py

```diff
@@ -328,9 +325,14 @@ def parse_config(config: dict) -> SamlConfig:
             SamlConfig: A custom config object for this module
         """
         # Parse config options and use defaults where necessary
-        # We use the 'or' syntax here as these options could be a valid value of None
+
+        # `config` can contain "mxid_source_attribute: None". Using `config.get`
```

Comment:
why would it contain that though?

I mean, what you've done isn't wrong, but I don't see why it would apply here any more than everywhere else that we read values from the config.

I'd just use `get` with a default.

Reply:
I was under the impression that commenting out `config:` was giving me `None` here for some reason, but I just tested it again and can't see the same failure mode, so it must've been some issue with the earlier crazy parsing in saml2_config.

Anyways, have changed to `gets` and everything seems quite happy now.